### DISCUSSION TITLE
Custom Fields - Selection returns array in REST API?

### DISCUSSION THREAD

**Discussion by ed-ud** [netbox-community/netbox#12418] [Help Wanted!]
Hi,

I have a Custom Field of type Selection (with comma-separated choice strings).  When I query an object that uses the custom field with one of the values, the field is actually returned in the JSON as an array instead of a simple string.  Is this the expected behavior?  This behavior makes it so I can't filter using this custom field with .../?cf_my_custom_field=stringchoice.

**Comment by candlerb** [comment]
* What Netbox version are you running?
* Can you show the full definition of the custom field? (Go to Custom Fields, select Export > All Data (CSV), then trim to just the heading + the custom field in question)
* Are you sure you didn't define the custom field as "Multiple Selection" instead of "Selection"?

It works for me with Netbox 3.4.10, for example:

```
      ...
      "custom_fields": {
        "snmp_module": "if_mib_v3"
      },
      ...
```

**Comment by ed-ud** [comment]
I'm running 3.4.6 currently.  Here is the definition...

```
Name,Content types,Label,Group name,Type,Required,Description,ID,Default,Search weight,Filter logic,UI visibility,Display weight,Choices,Created,Last updated
vcenter_cluster_ipam,ipam.prefix,IP Prefix VM Cluster Association,VMware,Selection,False,The VMware cluster this subnet is associated with (if any).,47,CHPPRD3,1000,Loose,Read/Write,100,"['CHPPRD3', 'DRPRD3']",2023-05-01 16:12,2023-05-01 16:24
```

**Comment by ed-ud** [comment]
On an object returned via the REST API:

```
...
            "custom_fields": {
                "vcenter_cluster_ipam": [
                    "CHPPRD3"
                ],
                "vpn_pool_name": "itnsssys"
            },
...
```

**Reply by candlerb** [reply]
That's very strange. What if you query the database row directly? e.g.

```
sudo -u postgres psql netbox
select id,custom_field_data from ipam_prefix where id=1;
```

Is it possible this field was originally Multiple Selection and then got changed to Selection?  I seem to recall a problem when this happened, and it's no longer allowed: #11431

Aside: if you end up changing this, the new "Object" custom field type will let you reference a VM cluster object in Netbox.

**Comment by ed-ud** [comment]
It is quite possible that the field type was changed.  I noticed mention of that new restriction on the releases page as well.  Given that, I expect this is a non-issue.  Thanks for the tip on using an Object->Cluster field... the only drawback there is I can't do a filter like /api/ipam/prefixes/?cf_vcenter_cluster_ipam=name and have to use the object ID of the cluster instead of the name... unless I'm missing something?

**Reply by candlerb** [reply]
I think you're right about the filtering. Many object types have slugs, but virtualization clusters don't. Even if they did, I don't know if custom field object filtering would allow you to match on the slug. Ref #12315

**Reply by ed-ud** [reply]
Yeah I think you're right... thanks Brian!
